### ISSUE TITLE
Officially support "Glossary" and smart terms within Glossary

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-ogc#245]
OGC documents have been confirmed to have both a "Terms and definitions" clause and a "Glossary" that is an annex.

The "Terms and definitions" clause will follow ISO practice, which allows:

* "Terms and definitions"
* "Terms, definitions and abbreviated terms"
* "Terms, definitions and symbols"
* "Terms, definitions, symbols and abbreviated terms"

With their internal subdivisions according to ISO practice.

The Glossary is an "informative" section for terms, which will contain informative terms, encoded in the same way as with the "Terms and definitions" clause. It is not confirmed whether it can contain symbols or abbreviated terms.

**Comment by ronaldtse** [comment]
Failure to reference a term:

```adoc
[appendix]
[heading='terms and definitions']
== Glossary

=== Standardization Target

An entity to which some requirements of a standard apply

[.source]
<<ogc08-131,clause=4.23>>
```

```adoc
== Conformance

This Standard defines a {{Conceptual Model}} which is independent of any encoding or 
formatting techniques. The {{Standardization Target,Standardization Targets}} for this 
Standard are:
```

Error:
```
AsciiDoc Input: (XML Line 000417): Error: Term reference in `term[standardization-target]` missing:                 "standardization-target" is not defined in document
```

**Comment by opoudjis** [comment]
Yes, `term:[]` references are currently restricted to real Terms & Definitions sections, so this OGC glossary annex will need to also be an explicit terms section.

**Comment by ronaldtse** [comment]
> this OGC glossary annex will need to also be an explicit terms section.

But this doesn't work?

```adoc
[appendix]
[heading='terms and definitions']
== Glossary
```

**Comment by opoudjis** [comment]
Because appendixes override headings.

**Comment by ronaldtse** [comment]
Got it, then there is further work needed to actually support the "Glossary".

**Comment by opoudjis** [comment]
In particular, as with NIST, glossaries within annexes need to be marked up as `<annex><terms>`, and the `<terms>` needs not to be treated as a separate subclause. This involves bringing across the relevant code from NIST.
